tests_pytest/common_tests/unit_tests/test_base_graph.py

```diff
@@ -0,0 +1,215 @@
+        exp_candidate_base2 = build_qc(a_nbits=4, a_enable=True, w_attr={'weight': (4, True)},
+                                        activation_quantization_fn=symmetric_selection_histogram,
+                                        activation_quantization_method=QuantizationMethod.SYMMETRIC)
+        
+        exp_actq_cfg_params_dict1 = {'quant_mode': ActivationQuantizationMode.FLN_QUANT,
+                                     'n_bits': 4,
+                                     'actq_params_fn': lut_kmeans_histogram,
+                                     'actq_method': QuantizationMethod.LUT_POT_QUANTIZER,
+                                     'signedness': Signedness.AUTO}
+        
+        exp_actq_cfg_params_dict2 = {'quant_mode':ActivationQuantizationMode.FLN_NO_QUANT,
+                                     'n_bits': 8,
+                                     'actq_params_fn': symmetric_selection_histogram,
+                                     'actq_method': QuantizationMethod.SYMMETRIC}
+
+        exp_actq_cfg_params_dict3 = {'quant_mode':ActivationQuantizationMode.QUANT,
+                                     'n_bits': 8,
+                                     'actq_params_fn': symmetric_selection_histogram,
+                                     'actq_method': QuantizationMethod.SYMMETRIC}
+
+        ### Expected candidates when transformed by the qcfg of FusingInfo
+        conv_1_qc_cfg = [create_exp_candidate_cfg(exp_candidate_base1, **exp_actq_cfg_params_dict1)]
+        ### Expected values when unchanged
+        relu_1_qc_cfg = [exp_candidate_base1]
+        ### Expected candidates when transformed by FusingInfo where qcfg is None
+        conv_2_qc_cfg = [create_exp_candidate_cfg(exp_candidate_base1, **exp_actq_cfg_params_dict2)]
+        ### Expected values when unchanged
+        tanh_qc_cfg   = [exp_candidate_base1]
+        ### Expected candidates with multiple configurations when transformed by the qcfg of FusingInfo
+        conv_3_qc_cfg = [create_exp_candidate_cfg(exp_candidate_base1, **exp_actq_cfg_params_dict1),
+                         create_exp_candidate_cfg(exp_candidate_base2, **exp_actq_cfg_params_dict1)]
+        ### Expected values when unchanged
+        relu_2_qc_cfg = [exp_candidate_base1, exp_candidate_base2]
+        ### Expected candidates when transformed with preserving set to True
+        flatten_qc_cfg = [create_exp_candidate_cfg(exp_candidate_base1, **exp_actq_cfg_params_dict3)]
+        ### Expected values when unchanged
+        linear_qc_cfg = [exp_candidate_base1, exp_candidate_base2]
+
+        self.exp_node_config_list = [conv_1_qc_cfg, relu_1_qc_cfg,
+                                     conv_2_qc_cfg, tanh_qc_cfg,
+                                     conv_3_qc_cfg, relu_2_qc_cfg,
+                                     flatten_qc_cfg, linear_qc_cfg]
+    
+    def check_candidates_activation_qcfg(self, candidates, exp_candidates):
```

Comment:
This check needs a little modification - first of all, if you are expecting an FLN quantization, check that only a single candidate is set for the node, if not an FLN - verify the number of expected candidates.
Then, we only care to verify the content of the candidate for an FLN node, so no need for the for loop in my opinion